policies.tf

```diff
@@ -52,3 +52,23 @@ data "aws_iam_policy_document" "task_execution_permissions" {
     ]
   }
 }
+
+# Task ssm privileges
+
+data "aws_iam_policy_document" "task_parameter_permissions" {
+  count = "${var.create ? 1 : 0}"
+
+  statement {
+    effect = "Allow"
+
+    resources = [
+      "*",
```

Comment:
We can output the role arn and allow user to attach custom policies, Policies intersections would take effect. updated.

Reply:
It is difficult to remember to add these intersected policies later. Since this policy controls access to SSM, let's specify the resources specifically in this policy.

Reply:
Add a

```
variable "task_container_environment_ssm_arns" {
  type = "list"
  description = "A list of the ARNs of the AWS SSM parameters representing our environment variables and secrets"
}
```

Reply:
Also, split the policy into 2 statements:

```
data "aws_iam_policy_document" "task_parameter_permissions" {
  statement {
    effect = "Allow"

    resources = ["${var.task_container_environment_ssm_arns}"]

    actions = [
      "ssm:GetParameters",
      "secretsmanager:GetSecretValue",
    ]
  }

  statement {
    effect = "Allow"

    resources = ["${var.ssm_kms_key_arn}"]

    actions = [
      "kms:Decrypt",
    ]
  }
}
```